httpclient/src/main/kotlin/tbdex/sdk/httpclient/TbdexHttpClient.kt

```diff
@@ -72,6 +75,42 @@ object TbdexHttpClient {
     }
   }
 
+  /**
+   * Fetches balances from a PFI.
+   *
+   * @param pfiDid The decentralized identifier of the PFI.
+   * @param requesterDid The decentralized identifier of the entity requesting the balances.
+   * @return A list of [Balance] matching the request.
+   * @throws TbdexResponseException for request or response errors.
+   */
+  fun getBalances(pfiDid: String, requesterDid: BearerDid): List<Balance> {
+    val pfiServiceEndpoint = getPfiServiceEndpoint(pfiDid)
+    val baseUrl = "$pfiServiceEndpoint/balances/"
+    val requestToken = RequestToken.generate(requesterDid, pfiDid)
+
+    val request = Request.Builder()
+      .url(baseUrl)
+      .addHeader(CONTENT_TYPE, JSON_HEADER)
+      .addHeader(AUTHORIZATION, "Bearer $requestToken")
+      .get()
+      .build()
+
+    val response: Response = client.newCall(request).execute()
+    when {
+      response.isSuccessful -> {
+        val responseString = response.body?.string()
+        // response body is an object with a data field
+        val jsonNode = jsonMapper.readTree(responseString)
+        return jsonNode.get("data").elements()
+          .asSequence()
+          .map { Balance.parse(it.toString()) }
```

Comment:
Nit/suggestion: Wrap this `Balance.parse()` in a `try/catch` that catches and re-throws as a `TbdexResponseException`.

Reply:
done!